xonsh/main.py

```diff
@@ -224,8 +224,8 @@ def main(argv=None):
         # run a script contained in a file
         path = os.path.abspath(os.path.expanduser(args.file))
         if os.path.isfile(path):
-            sys.argv = args.args
-            env['ARGS'] = [args.file] + args.args
+            sys.argv = [args.file] + args.args
+            env['ARGS'] = sys.argv[:] # $ARGS is not sys.argv
```

Comment:
Because there is no other `$ENV_VARIABLE` that is alias to something. Furthermore, one may modify `$ARGS`, reckoning to pass it to a subprocess, but not expecting `sys.argv` to be modified.